providers/dns/namecheap/namecheap.go

```diff
@@ -180,18 +180,20 @@ func (d *DNSProvider) CleanUp(domain, token, keyAuth string) error {
 
 	// Find the challenge TXT record and remove it if found.
 	var found bool
-	for i, h := range records {
+	newRecords := records[:0]
```

Comment:
Why not?

```suggestion
	newRecords := make([]Record, 0)
```

Reply:
It's a golang tricks to avoid allocation https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating

Reply:
in this case, `records` can be modify without any consequences.

Reply:
Neat trick! I'll keep that in mind.

Typically, I'll sacrifice performance to make something more readable (nothing is less performant than code that someone can't understand, especially those with side-effects!)